auth/cache.hh

```diff
@@ -42,17 +42,20 @@ public:
     };
 
     explicit cache(cql3::query_processor& qp) noexcept;
+    future<> ensure_initialized() noexcept;
     lw_shared_ptr<const role_record> get(const role_name_t& role) const noexcept;
     future<> load_all();
     future<> load_roles(std::unordered_set<role_name_t> roles);
     static bool includes_table(const table_id&) noexcept;
 
 private:
+    shared_promise<> _ensure_initialized;
```

Comment:
I don't love the shared_promise stuff. Is there no place where the starting order is unambiguous?

I realize the shared_promise was triggered by a review comment, I don't want to enter a loop, but I want to understand why we can't have a single order.

Also, are the implications for shutdown order?

Reply:
But there is one exception, when nodes restart in recovery mode. In such case it seems snapshot is not loaded.